sdk/core/core-http/src/policies/proxyPolicy.ts

```diff
@@ -12,23 +12,54 @@ import { ProxySettings } from "../serviceClient";
 import { WebResourceLike } from "../webResource";
 import { Constants } from "../util/constants";
 import { URLBuilder } from "../url";
+import { getEnvironmentValue } from "../util/utils";
+
+let noProxyList: any = null;
 
 function loadEnvironmentProxyValue(): string | undefined {
   if (!process) {
     return undefined;
   }
 
-  if (process.env[Constants.HTTPS_PROXY]) {
-    return process.env[Constants.HTTPS_PROXY];
-  } else if (process.env[Constants.HTTPS_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTPS_PROXY.toLowerCase()];
-  } else if (process.env[Constants.HTTP_PROXY]) {
-    return process.env[Constants.HTTP_PROXY];
-  } else if (process.env[Constants.HTTP_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTP_PROXY.toLowerCase()];
+  let httpsProxy =
+    getEnvironmentValue(Constants.HTTPS_PROXY) || getEnvironmentValue(Constants.ALL_PROXY);
+  let httpProxy =
+    getEnvironmentValue(Constants.HTTP_PROXY) || getEnvironmentValue(Constants.ALL_PROXY);
+
+  return httpsProxy || httpProxy;
+}
+
+// Check whether uri match noProxyList. If match, won't set proxy settings while sending request to the uri.
+function isBypassed(uri: string) {
```

Comment:
Do you have unit tests that cover this method?

Reply:
@jongio isBypassed is call in sendrequest, we have add more scenarios to test sendrequest.

Reply:
Thanks